homeassistant/components/ecovacs.py

```diff
@@ -0,0 +1,95 @@
+"""Parent component for Ecovacs Deebot vacuums.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/ecovacs/
+"""
+
+import logging
+import random
+import string
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers import discovery
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD, \
+    EVENT_HOMEASSISTANT_STOP
+
+REQUIREMENTS = ['sucks==0.9.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "ecovacs"
+
+CONF_COUNTRY = "country"
+CONF_CONTINENT = "continent"
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Required(CONF_COUNTRY): cv.string,
+        vol.Required(CONF_CONTINENT): cv.string,
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+ECOVACS_DEVICES = "ecovacs_devices"
+
+# Generate a random device ID on each bootup
+ECOVACS_API_DEVICEID = ''.join(
+    random.choice(string.ascii_uppercase + string.digits) for _ in range(8)
+)
+
+
+def setup(hass, config):
+    """Set up the Ecovacs component."""
+    _LOGGER.info("Creating new Ecovacs component")
+
+    if ECOVACS_DEVICES not in hass.data:
+        hass.data[ECOVACS_DEVICES] = []
+
+    from sucks import EcoVacsAPI, VacBot
+
+    # Convenient hack for debugging to pipe sucks logging to the Hass logger
+    if _LOGGER.getEffectiveLevel() <= logging.DEBUG:
+        import sucks
+        sucks.logging = _LOGGER
```

Comment:
It would be good if the sucks library could be updated to use a logger per module, as is recommended.

https://docs.python.org/3/library/logging.html#logger-objects

Reply:
Yep, you're right. Done.